group-coordinator/src/main/java/org/apache/kafka/coordinator/group/consumer/TargetAssignmentBuilder.java

```diff
@@ -244,16 +244,21 @@ public TargetAssignmentResult build() throws PartitionAssignorException {
         });
 
         // Prepare the topic metadata.
-        Map<Uuid, AssignmentTopicMetadata> topics = new HashMap<>();
+        Map<Uuid, PartitionMetadata> topicMetadataMap = new HashMap<>();
         subscriptionMetadata.forEach((topicName, topicMetadata) ->
-            topics.put(topicMetadata.id(), new AssignmentTopicMetadata(topicMetadata.numPartitions()))
+            topicMetadataMap.put(
+                topicMetadata.id(),
+                new PartitionMetadata(topicMetadata.partitionRacks())
+            )
         );
 
         // Compute the assignment.
-        GroupAssignment newGroupAssignment = assignor.assign(new AssignmentSpec(
-            Collections.unmodifiableMap(memberSpecs),
-            Collections.unmodifiableMap(topics)
-        ));
+        GroupAssignment newGroupAssignment = assignor.assign(
+            new AssignmentSpec(
+                Collections.unmodifiableMap(memberSpecs)
+            ),
```

Comment:
nit: `new AssignmentSpec(Collections.unmodifiableMap(memberSpecs)),` as it fits on one line.